src/linker/Linker/MessageContainer.cs

```diff
@@ -134,8 +134,9 @@ private static MessageContainer CreateWarningMessageContainer (LinkContext conte
 
 			if (subcategory == MessageSubCategory.TrimAnalysis) {
 				Debug.Assert (origin.MemberDefinition != null);
-				var assembly = origin.MemberDefinition?.DeclaringType.Module.Assembly;
-				var assemblyName = assembly?.Name.Name;
+				var declaringType = origin.MemberDefinition?.DeclaringType ?? (origin.MemberDefinition as TypeDefinition);
+				var assembly = declaringType?.Module?.Assembly;
```

Comment:
Do these need `?` for every dereference? I thought the first `?` would cause the whole thing to return null if declaringType was null.

Reply:
You're right... I was just... "let's git rid of null ref" :wink: